Marlin/src/HAL/HAL_LPC1768/LPC1768_PWM.cpp

```diff
@@ -134,35 +110,13 @@ PWM_map ISR_table[NUM_PWMS] = PWM_MAP_INIT;
 #define P2_04_PWM_channel  5  // D9
 #define P2_05_PWM_channel  6  // D10
 
-// used to keep track of which Match Registers have been used and if they will be used by the
-// PWM1 module to directly control the pin or will be used to generate an interrupt
-typedef struct {                    // status of PWM1 channel
-  uint8_t             map_used;     // 0 - this MR register not used/assigned
-  uint8_t             map_PWM_INT;  // 0 - available for interrupts, 1 - in use by PWM
-  pin_t               map_PWM_PIN;  // pin for this PwM1 controlled pin / port
-  volatile uint32_t*  MR_register;  // address of the MR register for this PWM1 channel
-  uint32_t            PCR_bit;      // PCR register bit to enable PWM1 control of this pin
-                                    // 0 - don't switch to PWM1 direct control
-  volatile uint32_t*  PINSEL_reg;   // PINSEL register
-  uint32_t            PINSEL_bits;  // PINSEL register bits to set pin mode to PWM1 control
-} MR_map;
-
-MR_map map_MR[NUM_PWMS];
-
-void LPC1768_PWM_update_map_MR(void) {
-  map_MR[0] = { 0, (uint8_t) (LPC_PWM1->PCR & _BV(8 + P1_18_PWM_channel) ? 1 : 0), P1_18, &LPC_PWM1->MR1, 0, 0, 0 };
-  map_MR[1] = { 0, (uint8_t) (LPC_PWM1->PCR & _BV(8 + P1_20_PWM_channel) ? 1 : 0), P1_20, &LPC_PWM1->MR2, 0, 0, 0 };
-  map_MR[2] = { 0, (uint8_t) (LPC_PWM1->PCR & _BV(8 + P1_21_PWM_channel) ? 1 : 0), P1_21, &LPC_PWM1->MR3, 0, 0, 0 };
-  map_MR[3] = {
-    #if MB(MKS_SBASE)
-      0, (uint8_t) (LPC_PWM1->PCR & _BV(8 + P1_23_PWM_channel) ? 1 : 0), P1_23, &LPC_PWM1->MR4, 0, 0, 0
-    #else
-      0, 0, P_NC, &LPC_PWM1->MR4, 0, 0, 0
-    #endif
-  };
-  map_MR[4] = { 0, (uint8_t) (LPC_PWM1->PCR & _BV(8 + P2_04_PWM_channel) ? 1 : 0), P2_04, &LPC_PWM1->MR5, 0, 0, 0 };
-  map_MR[5] = { 0, (uint8_t) (LPC_PWM1->PCR & _BV(8 + P2_05_PWM_channel) ? 1 : 0), P2_05, &LPC_PWM1->MR6, 0, 0, 0 };
-}
+typedef struct {
+  uint32_t min;
+  uint32_t max;
+  bool     assigned;
+} table_direct;
+
+table_direct direct_table[6] = {0};
```

Comment:
It's hard to tell what this "0" refers to. Since the loader inits the contents of all static buffers to 0 anyway, it may be ok to just leave it off.